Subject: remove "opt" plugin ("ctr install")
Forwarded: no; this is very intentional upstream behavior (https://github.com/containerd/containerd/commit/1537f31381b946f395348d9891602a1e4a2c1070) that we just disagree with having/supporting

diff --git a/cmd/containerd/builtins/builtins.go b/cmd/containerd/builtins/builtins.go
index 9d6e180a4..16fdce720 100644
--- a/cmd/containerd/builtins/builtins.go
+++ b/cmd/containerd/builtins/builtins.go
@@ -38,7 +38,6 @@ import (
 	_ "github.com/containerd/containerd/services/introspection"
 	_ "github.com/containerd/containerd/services/leases"
 	_ "github.com/containerd/containerd/services/namespaces"
-	_ "github.com/containerd/containerd/services/opt"
 	_ "github.com/containerd/containerd/services/sandbox"
 	_ "github.com/containerd/containerd/services/snapshots"
 	_ "github.com/containerd/containerd/services/streaming"
