Description: Remove LimitNOFILE from containerd.service
Author: Tianon <tianon@debian.org> (& Brennan Kinney)
Applied-Upstream: 2.0+
Forwarded: https://github.com/containerd/containerd/pull/8924

diff --git a/containerd.service b/containerd.service
index 4ee5d9312..15556d2dd 100644
--- a/containerd.service
+++ b/containerd.service
@@ -30,7 +30,6 @@ RestartSec=5
 # in the kernel. We recommend using cgroups to do container-local accounting.
 LimitNPROC=infinity
 LimitCORE=infinity
-LimitNOFILE=infinity
 # Comment TasksMax if your systemd version does not supports it.
 # Only systemd 226 and above support this version.
 TasksMax=infinity
