Description: fix service ExecStart path
Author: Tianon Gravi <tianon@debian.org>
Forwarded: not-needed

diff --git a/containerd.service b/containerd.service
index 5f67110ab..0f45ad0cd 100644
--- a/containerd.service
+++ b/containerd.service
@@ -5,7 +5,7 @@ After=network.target local-fs.target
 
 [Service]
 ExecStartPre=-/sbin/modprobe overlay
-ExecStart=/usr/local/bin/containerd
+ExecStart=/usr/bin/containerd
 
 Type=notify
 Delegate=yes
