Description: fix service ExecStart path
Author: Tianon Gravi <tianon@debian.org>
Forwarded: not-needed

diff --git a/containerd.service b/containerd.service
index dcdb9f414..3424f09a8 100644
--- a/containerd.service
+++ b/containerd.service
@@ -21,7 +21,7 @@ After=network.target local-fs.target dbus.service
 #uncomment to enable the experimental sbservice (sandboxed) version of containerd/cri integration
 #Environment="ENABLE_CRI_SANDBOXES=sandboxed"
 ExecStartPre=-/sbin/modprobe overlay
-ExecStart=/usr/local/bin/containerd
+ExecStart=/usr/bin/containerd
 
 Type=notify
 Delegate=yes
