Origin: https://github.com/docker/buildx/pull/1886
Applied-Upstream: 0.12+
From: CrazyMax <crazy-max@users.noreply.github.com>
Date: Wed, 21 Jun 2023 00:13:44 +0200
Subject: build: prefer local image resolution for docker driver

diff --git a/build/build.go b/build/build.go
index 21b502a07f..b480187a33 100644
--- a/build/build.go
+++ b/build/build.go
@@ -593,7 +593,10 @@ func toSolveOpt(ctx context.Context, node builder.Node, multiDriver bool, opt Op
 	}
 
 	if opt.Pull {
-		so.FrontendAttrs["image-resolve-mode"] = "pull"
+		so.FrontendAttrs["image-resolve-mode"] = pb.AttrImageResolveModeForcePull
+	} else if nodeDriver.IsMobyDriver() {
+		// moby driver always resolves local images by default
+		so.FrontendAttrs["image-resolve-mode"] = pb.AttrImageResolveModePreferLocal
 	}
 	if opt.Target != "" {
 		so.FrontendAttrs["target"] = opt.Target
